microservices/front-end/src/components/MainQuestionBank.tsx

```diff
@@ -81,13 +88,32 @@ const QuestionBank: React.FC = () => {
   const [addError, setAddError] = useState<string | null>(null);
   const [updateError, setUpdateError] = useState<string | null>(null);
 
+  // maintain filter & sort states
+  const [selectedDifficulty, setSelectedDifficulty] = useState<string>("All");
+  const [filteredCategory, setFilteredCategory] = useState<string>("All");
+  const [attemptedFilter, setAttemptedFilter] = React.useState<string>("All");
+  const [searchQuery, setSearchQuery] = useState<string>("");
+  const [sortBy, setSortBy] = useState<string>("");
```

Comment:
Most of the default values for React `useState` are `"All"` as expected. Did you forget to pass in the React value into the `QuestionFilter.tsx` component?

(See my other comment about why we should not have `""` (empty string) as default values)